engine/runtime/src/main/java/org/enso/interpreter/node/ClosureRootNode.java

```diff
@@ -54,7 +54,7 @@ public class ClosureRootNode extends EnsoRootNode {
   public static ClosureRootNode build(
       EnsoLanguage language,
       LocalScope localScope,
-      ModuleScope moduleScope,
+      ModuleScope.Builder moduleScope,
```

Comment:
I believe only `IrToTruffle` should use `ModuleScope.Builder` for its purposes. All the runtime nodes should have only immutable `ModuleScope`. Is that possible? Having `ModuleScope.Builder` as fields in some truffle nodes diminishes the whole idea of this PR, since we will still be able to mutate the underlying builder at runtime, which should be restricted. No?